src/rgw/rgw_lc_s3.cc

```diff
@@ -180,6 +220,8 @@ XMLObj *RGWLCXMLParser_S3::alloc_obj(const char *el)
     obj = new LCRule_S3();
   } else if (strcmp(el, "ID") == 0) {
     obj = new LCID_S3();
+  } else if (strcmp(el, "Filter") == 0) {
```

Comment:
With this many branches and strcmp() calls, I start to wonder if it's time to rethink the approach (maybe it's not quite the time today, but...).